### PR TITLE
Foxy remove not needed features

### PR DESCRIPTION
**Pull request by carlossvg**
This MR simplifies the complexity of the project removing not needed features:

- Remove "OSRF memory tools" usage. ros2_tracing can be used instead.
- Remove statistics. ROS 2 "Topic Statistics" will be used instead.
- Remove simulation thread
- Remove logging macros
- Add CI support for Foxy builds

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ Note the force is applied continuously until a different disturbance force is se
 
 #### Managed nodes
 
-The main nodes used in the demo `/pendulum_controller` and `/pendulum_driver` are [managed nodes](https://design.ros2.org/articles/node_lifecycle.html). That is, they inherit from a Lifecycle node and it is posible to control the node state.
+Thmain nodes used in the demo `/pendulum_controller` and `/pendulum_driver` are [managed nodes](https://design.ros2.org/articles/node_lifecycle.html). That is, they inherit from a Lifecycle node and it is posible to control the node state.
```
Comment (LanderU) [inline README.md]:
```suggestion
The main nodes used in the demo `/pendulum_controller` and `/pendulum_driver` are [managed nodes](https://design.ros2.org/articles/node_lifecycle.html). That is, they inherit from a Lifecycle node and it is posible to control the node state.
```

**Comment by LanderU** [comment]
@carlossvg thanks for the changes. Also, I think this changes will fix the uncrustify issues on https://github.com/ros2-realtime-demo/pendulum/issues/21#issuecomment-660629644.

**Comment by LanderU** [comment]
@carlossvg confirmed this branch solves the issue with uncrustify. Only one thing, shall we modify this deprecated call?

```bash
--- stderr: pendulum_demo
/root/ros2_symlink/src/pendulum/pendulum_demo/src/pendulum_controller_standalone.cpp: In function ‘int main(int, char**)’:
/root/ros2_symlink/src/pendulum/pendulum_demo/src/pendulum_controller_standalone.cpp:170:34: warning: ‘using ExecutorArgs = struct rclcpp::ExecutorOptions’ is deprecated: use rclcpp::ExecutorOptions instead [-Wdeprecated-declarations]
  170 |   rclcpp::executor::ExecutorArgs exec_args;
      |                                  ^~~~~~~~~
In file included from /opt/ros/foxy/include/rclcpp/executor.hpp:33,
                 from /opt/ros/foxy/include/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/foxy/include/rclcpp/executors.hpp:21,
                 from /opt/ros/foxy/include/rclcpp/rclcpp.hpp:146,
                 from /root/ros2_symlink/src/pendulum/pendulum_demo/src/pendulum_controller_standalone.cpp:27:
/opt/ros/foxy/include/rclcpp/executor_options.hpp:44:7: note: declared here
   44 | using ExecutorArgs [[deprecated("use rclcpp::ExecutorOptions instead")]] = ExecutorOptions;
      |       ^~~~~~~~~~~~
/root/ros2_symlink/src/pendulum/pendulum_demo/src/pendulum_driver_standalone.cpp: In function ‘int main(int, char**)’:
/root/ros2_symlink/src/pendulum/pendulum_demo/src/pendulum_driver_standalone.cpp:142:34: warning: ‘using ExecutorArgs = struct rclcpp::ExecutorOptions’ is deprecated: use rclcpp::ExecutorOptions instead [-Wdeprecated-declarations]
  142 |   rclcpp::executor::ExecutorArgs exec_args;
      |                                  ^~~~~~~~~
In file included from /opt/ros/foxy/include/rclcpp/executor.hpp:33,
                 from /opt/ros/foxy/include/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/foxy/include/rclcpp/executors.hpp:21,
                 from /opt/ros/foxy/include/rclcpp/rclcpp.hpp:146,
                 from /root/ros2_symlink/src/pendulum/pendulum_demo/src/pendulum_driver_standalone.cpp:27:
/opt/ros/foxy/include/rclcpp/executor_options.hpp:44:7: note: declared here
   44 | using ExecutorArgs [[deprecated("use rclcpp::ExecutorOptions instead")]] = ExecutorOptions;
      |       ^~~~~~~~~~~~
/root/ros2_symlink/src/pendulum/pendulum_demo/src/pendulum_demo.cpp: In function ‘int main(int, char**)’:
/root/ros2_symlink/src/pendulum/pendulum_demo/src/pendulum_demo.cpp:186:34: warning: ‘using ExecutorArgs = struct rclcpp::ExecutorOptions’ is deprecated: use rclcpp::ExecutorOptions instead [-Wdeprecated-declarations]
  186 |   rclcpp::executor::ExecutorArgs exec_args;
      |                                  ^~~~~~~~~
In file included from /opt/ros/foxy/include/rclcpp/executor.hpp:33,
                 from /opt/ros/foxy/include/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/foxy/include/rclcpp/executors.hpp:21,
                 from /opt/ros/foxy/include/rclcpp/rclcpp.hpp:146,
                 from /root/ros2_symlink/src/pendulum/pendulum_demo/src/pendulum_demo.cpp:27:
/opt/ros/foxy/include/rclcpp/executor_options.hpp:44:7: note: declared here
   44 | using ExecutorArgs [[deprecated("use rclcpp::ExecutorOptions instead")]] = ExecutorOptions;
      |       ^~~~~~~~~~~~
---
```

**Comment by carlossvg** [comment]
> @carlossvg confirmed this branch solves the issue with uncrustify. Only one thing, shall we modify this deprecated call?

Fixed

**Review by LanderU — APPROVED** [review]
LGTM!

**Pull request merged**